From 682ab9370c1e6ec6ae8fe78971bd04d0f431ad36 Mon Sep 17 00:00:00 2001
From: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Date: Thu, 4 Oct 2018 11:23:43 -0700
Subject: [PATCH 0436/1852] xilinx: v4l: dma: Remove unused function definition

commit f9629cd935b66dee2013eb9e0a7db417a2eeb267 from
https://github.com/Xilinx/linux-xlnx.git

This patch removes unused function definition xvip_get_entity_sink().

Signed-off-by: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-dma.c | 29 ----------------------
 1 file changed, 29 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-dma.c b/drivers/media/platform/xilinx/xilinx-dma.c
index 0fae7570cdda..1b7068bcf45c 100644
--- a/drivers/media/platform/xilinx/xilinx-dma.c
+++ b/drivers/media/platform/xilinx/xilinx-dma.c
@@ -93,35 +93,6 @@ static int xvip_dma_verify_format(struct xvip_dma *dma)
  * Pipeline Stream Management
  */
 
-/* Get the sink pad internally connected to a source pad in the given entity. */
-static struct media_pad *xvip_get_entity_sink(struct media_entity *entity,
-					      struct media_pad *source)
-{
-	unsigned int i;
-
-	/* The source pad can be NULL when the entity has no source pad. Return
-	 * the first pad in that case, guaranteed to be a sink pad.
-	 */
-	if (!source)
-		return &entity->pads[0];
-
-	/* Iterates through the pads to find a connected sink pad. */
-	for (i = 0; i < entity->num_pads; ++i) {
-		struct media_pad *sink = &entity->pads[i];
-
-		if (!(sink->flags & MEDIA_PAD_FL_SINK))
-			continue;
-
-		if (sink == source)
-			continue;
-
-		if (media_entity_has_route(entity, sink->index, source->index))
-			return sink;
-	}
-
-	return NULL;
-}
-
 /**
  * xvip_pipeline_start_stop - Start ot stop streaming on a pipeline
  * @xdev: Composite video device
-- 
2.31.1

